docs/shrc.md

```diff
@@ -22,37 +22,116 @@
+- If class data sharing fails, the VM still starts without printing any errors.
+
+Shared class behavior is controlled by specifying the [-Xshareclasses](xshareclasses.md) option on the command line when you start your application. The default settings are equivalent to specifying the following suboptions:
+
+```
+-Xshareclasses:bootClassesOnly,nonFatal,silent
+```
+
+You can enable class data sharing for non-bootstrap classes as well as bootstrap classes, by omitting the `bootClassesOnly` suboption. You can also disable all class data sharing by using the `none` suboption.
+
+Further suboptions are available to configure a cache at startup, including name, location, and startup size. You can also use cache utilities to manage a shared classes cache after it is initialized and in use.
+
+A shared classes cache can be *persistent* or *non-persistent* according to the following definition:
+
+- persistent caches are written to memory-mapped files and remain in place, even after a system is rebooted.
+- non-persistent caches exist in shared memory and are automatically removed when the operating system is restarted.
+
+By default, a shared classes cache is persistent, except on the z/OS platform, where persistent caches are not supported.
+If you are using a non-persistent cache, you can use a cache utility to create a snapshot of the cache, which can be reinitialized after a reboot. For more information see [Saving a non-persistent shared classes cache](#saving-a-non-persistent-shared-classes-cache).
+
+If you have multiple VMs and you do not change the default shared classes behavior, the VMs share a single default cache, assuming that the VMs are from a single Java installation. If the VMs are from different Java installations, the cache might be deleted and re-created.
+
+For a set of best practices when using class data sharing, see [Creating a shared classes cache](#creating-a-shared-classes-cache).
+
+
+## Class data sharing operations
+
+When a VM loads a class and the class loader is enabled for class sharing, the VM looks in the shared classes cache to see if the class is already present. If the class is present and the classpath or URL to load the class is a match, the VM loads the class from the cache. Otherwise, it loads the class from the filesystem and writes it into the cache.
+
+The VM detects filesystem updates by storing timestamp values into the cache and comparing the cached values with actual values on each class load operation. In this way, the VM detects when a class might be invalidated and can mark the class as *stale*. These operations happen transparently when classes are loaded, so users can modify and update as many classes as they like during the lifetime of a shared class cache, knowing that the correct classes are always loaded. Stale classes are *redeemed* if the same class is subsequently fetched by the class loader from another VM and checked against the stale class in the cache.
```

Comment:
"on each class load operation" isn't correct, unless loading classes from a directory. For a jar file, it's checked once when opened, unless https://www.eclipse.org/openj9/docs/xshareclasses/#checkurltimestamps is used.